File: packages/features/feature_learning/lib/src/learning/components/cards/srs_learning_card.dart
import 'dart:ui';
import 'package:flutter/material.dart';
import 'package:core_designsystem/core_designsystem.dart';
import 'package:core_domain/core_domain.dart';
import 'package:core_ui/core_ui.dart';

enum CardBadge {
  fresh,
  review,
  relearn;

  String get text {
    switch (this) {
      case CardBadge.fresh: return '新学';
      case CardBadge.review: return '复习';
      case CardBadge.relearn: return '重学';
    }
  }

  Color getBgColor(bool isDark) {
    if (isDark) {
      return switch (this) {
        CardBadge.fresh => const Color(0xFF1E3A8A),
        CardBadge.review => const Color(0xFF14532D),
        CardBadge.relearn => const Color(0xFF7C2D12),
      };
    }
    return switch (this) {
      CardBadge.fresh => const Color(0xFFE0EDFF),
      CardBadge.review => const Color(0xFFDCFCE7),
      CardBadge.relearn => const Color(0xFFFFEDD5),
    };
  }

  Color getTextColor(bool isDark) {
    if (isDark) {
      return switch (this) {
        CardBadge.fresh => const Color(0xFFBFDBFE),
        CardBadge.review => const Color(0xFFBBF7D0),
        CardBadge.relearn => const Color(0xFFFED7AA),
      };
    }
    return switch (this) {
      CardBadge.fresh => const Color(0xFF1D4ED8),
      CardBadge.review => const Color(0xFF166534),
      CardBadge.relearn => const Color(0xFF9A3412),
    };
  }
}

class SRSLearningCard extends StatelessWidget {
  const SRSLearningCard({
    super.key,
    required this.word,
    required this.isAnswerShown,
    this.badge,
    this.onSpeakWord,
    this.onSpeakExample,
    this.onPracticeClick,
    this.playingAudioId,
  });

  final Word word;
  final bool isAnswerShown;
  final CardBadge? badge;
  final VoidCallback? onSpeakWord;
  final Function(String japanese, String chinese, String id)? onSpeakExample;
  final VoidCallback? onPracticeClick;
  final String? playingAudioId;

  @override
  Widget build(BuildContext context) {
    final isDark = Theme.of(context).brightness == Brightness.dark;

    final cardBackground = isDark ? NemoColors.surfaceCardDark : NemoColors.surfaceCard;
    final borderColor = isDark ? Colors.white.withValues(alpha: 0.1) : Colors.black.withValues(alpha: 0.05);
    final shadowColor = isDark ? Colors.black.withValues(alpha: 0.4) : Colors.black.withValues(alpha: 0.03);

    return SingleChildScrollView(
      physics: const BouncingScrollPhysics(),
      padding: const EdgeInsets.only(bottom: 120, left: 16, right: 16),
      child: Column(
        children: [
          const SizedBox(height: 16),
          // --- Question Area ---
          _QuestionBox(
            word: word,
            isAnswerShown: isAnswerShown,
            badge: badge,
            isDark: isDark,
            backgroundColor: cardBackground,
            borderColor: borderColor,
            shadowColor: shadowColor,
          ),

          const SizedBox(height: 16),

          // --- Sticker or Answer Area ---
          AnimatedSwitcher(
            duration: const Duration(milliseconds: 300),
            transitionBuilder: (child, animation) {
              return FadeTransition(
                opacity: animation,
                child: SlideTransition(
                  position: Tween<Offset>(
                    begin: const Offset(0, 0.1),
                    end: Offset.zero,
                  ).animate(animation),
                  child: child,
                ),
              );
            },
            child: isAnswerShown
                ? _AnswerBox(
                    key: const ValueKey('answer'),
                    word: word,
                    isDark: isDark,
                    backgroundColor: cardBackground,
                    borderColor: borderColor,
                    shadowColor: shadowColor,
                    onSpeakWord: onSpeakWord,
                    onSpeakExample: onSpeakExample,
                    onPracticeClick: onPracticeClick,
                    playingAudioId: playingAudioId,
                  )
                : _StickerBox(
                    key: const ValueKey('sticker'),
                    wordId: word.id,
                  ),
          ),
        ],
      ),
    );
  }
}

class _QuestionBox extends StatelessWidget {
  const _QuestionBox({
    required this.word,
    required this.isAnswerShown,
    this.badge,
    required this.isDark,
    required this.backgroundColor,
    required this.borderColor,
    required this.shadowColor,
  });

  final Word word;
  final bool isAnswerShown;
  final CardBadge? badge;
  final bool isDark;
  final Color backgroundColor;
  final Color borderColor;
  final Color shadowColor;

  @override
  Widget build(BuildContext context) {
    final primaryTextColor = isDark ? NemoColors.darkTextPrimary : NemoColors.gray900;

    return Container(
      width: double.infinity,
      decoration: BoxDecoration(
        color: backgroundColor,
        borderRadius: BorderRadius.circular(26),
        border: Border.all(color: borderColor, width: 0.5),
        boxShadow: [
          BoxShadow(
            color: shadowColor,
            blurRadius: 10,
            offset: const Offset(0, 4),
          ),
        ],
      ),
      padding: const EdgeInsets.symmetric(vertical: 32, horizontal: 16),
      child: Stack(
        clipBehavior: Clip.none,
        children: [
          SizedBox(
            width: double.infinity,
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.center,
              mainAxisSize: MainAxisSize.min,
              children: [
                // Japanese
                NemoFuriganaText(
                  text: word.japanese,
                  textAlign: TextAlign.center,
                  baseTextStyle: TextStyle(
                    fontSize: 48,
                    fontWeight: FontWeight.w800, // ExtraBold
                    letterSpacing: -1,
                    height: 1.1,
                    color: primaryTextColor,
                  ),
                ),
                const SizedBox(height: 8),
                // Hiragana (Blurred if hidden)
                _BlurredText(
                  text: word.hiragana,
                  isBlurred: !isAnswerShown,
                  style: TextStyle(
                    fontSize: 20,
                    fontWeight: FontWeight.w500,
                    color: isAnswerShown ? NemoColors.blue600 : (isDark ? Colors.white.withValues(alpha: 0.2) : NemoColors.gray300),
                  ),
                ),
              ],
            ),
          ),
          if (badge != null)
            Positioned(
              top: -8,
              right: -4,
              child: Container(
                padding: const EdgeInsets.symmetric(horizontal: 14, vertical: 6),
                decoration: BoxDecoration(
                  color: badge!.getBgColor(isDark),
                  borderRadius: BorderRadius.circular(20),
                ),
                child: Text(
                  badge!.text,
                  style: TextStyle(
                    color: badge!.getTextColor(isDark),
                    fontSize: 12,
                    fontWeight: FontWeight.w900,
                    letterSpacing: 1.2,
                  ),
                ),
              ),
            ),
        ],
      ),
    );
  }
}

class _BlurredText extends StatelessWidget {
  const _BlurredText({
    required this.text,
    required this.isBlurred,
    required this.style,
  });

  final String text;
  final bool isBlurred;
  final TextStyle style;

  @override
  Widget build(BuildContext context) {
    if (!isBlurred) {
      return Text(text, style: style);
    }
    return ImageFiltered(
      imageFilter: ImageFilter.blur(sigmaX: 8, sigmaY: 8),
      child: Text(text, style: style),
    );
  }
}

class _StickerBox extends StatelessWidget {
  const _StickerBox({super.key, required this.wordId});
  final String wordId;

  @override
  Widget build(BuildContext context) {
    return Container(
      width: double.infinity,
      height: 320,
      padding: const EdgeInsets.symmetric(vertical: 24),
      alignment: Alignment.center,
      child: Opacity(
        opacity: 0.1,
        child: Column(
          mainAxisSize: MainAxisSize.min,
          children: [
            const Icon(Icons.auto_awesome_rounded, size: 80, color: NemoColors.textMain),
            const SizedBox(height: 16),
            Text('STICKER_${wordId.hashCode % 25}', style: const TextStyle(fontWeight: FontWeight.w900, color: NemoColors.textMain)),
          ],
        ),
      ),
    );
  }
}

class _AnswerBox extends StatelessWidget {
  const _AnswerBox({
    super.key,
    required this.word,
    required this.isDark,
    required this.backgroundColor,
    required this.borderColor,
    required this.shadowColor,
    this.onSpeakWord,
    this.onSpeakExample,
    this.onPracticeClick,
    this.playingAudioId,
  });

  final Word word;
  final bool isDark;
  final Color backgroundColor;
  final Color borderColor;
  final Color shadowColor;
  final VoidCallback? onSpeakWord;
  final Function(String japanese, String chinese, String id)? onSpeakExample;
  final VoidCallback? onPracticeClick;
  final String? playingAudioId;

  @override
  Widget build(BuildContext context) {
    final practiceColor = _getColorForWord(word.id.hashCode);
    final practiceBgColor = practiceColor.withValues(alpha: isDark ? 0.2 : 0.1);

    final primaryTextColor = isDark ? NemoColors.darkTextPrimary : NemoColors.gray900;
    final secondaryTextColor = isDark ? NemoColors.gray400 : NemoColors.gray400;
    final dividerColor = isDark ? Colors.white.withValues(alpha: 0.1) : NemoColors.gray100;

    return Container(
      width: double.infinity,
      decoration: BoxDecoration(
        color: backgroundColor.withValues(alpha: 0.95),
        borderRadius: BorderRadius.circular(26),
        border: Border.all(color: borderColor, width: 0.5),
        boxShadow: [
          BoxShadow(
            color: shadowColor,
            blurRadius: 10,
            offset: const Offset(0, 4),
          ),
        ],
      ),
      padding: const EdgeInsets.all(24),
      child: Stack(
        children: [
          Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              // POS Tag
              if (word.pos != null)
                _PosPill(text: word.pos!, isDark: isDark),

              // Meaning Section
              _Label('含义', color: secondaryTextColor),
              const SizedBox(height: 4),
              Text(
                word.chinese,
                style: TextStyle(
                  fontSize: 24,
                  fontWeight: FontWeight.bold,
                  color: primaryTextColor,
                  height: 1.2,
                ),
              ),
              const SizedBox(height: 16),
              Divider(height: 1, color: dividerColor),
              const SizedBox(height: 16),

              // Example Section
              _Label('例句', color: secondaryTextColor),
              const SizedBox(height: 12),
              ...word.examples.asMap().entries.map((entry) {
                final ex = entry.value;
                final id = 'ex${entry.key}_${word.id}';
                return _ExampleRow(
                  japanese: ex.japanese,
                  chinese: ex.chinese,
                  id: id,
                  onSpeak: onSpeakExample,
                  isPlaying: playingAudioId == id,
                  primaryTextColor: primaryTextColor,
                  secondaryTextColor: secondaryTextColor,
                );
              }),
            ],
          ),

          // Top Right Buttons
          Positioned(
            top: 0,
            right: 0,
            child: Row(
              mainAxisSize: MainAxisSize.min,
              children: [
                if (onSpeakWord != null)
                  _SpeakerButton(
                    onPressed: onSpeakWord!,
                    isPlaying: playingAudioId == 'word',
                    color: practiceColor,
                    backgroundColor: practiceBgColor,
                  ),
                const SizedBox(width: 8),
                if (onPracticeClick != null)
                  _PracticeButton(
                    onPressed: onPracticeClick!,
                    color: practiceColor,
                    backgroundColor: practiceBgColor,
                  ),
              ],
            ),
          ),
        ],
      ),
    );
  }

  Color _getColorForWord(int id) {
    final colors = [
      const Color(0xFF007AFF), // Blue
      const Color(0xFF34C759), // Green
      const Color(0xFFFF9500), // Orange
      const Color(0xFFFF2D55), // Pink
      const Color(0xFF5856D6), // Indigo
      const Color(0xFFAF52DE), // Purple
      const Color(0xFFFF3B30), // Red
      const Color(0xFF00C7BE), // Mint
    ];
    return colors[id.abs() % colors.length];
  }
}

class _PosPill extends StatelessWidget {
  const _PosPill({required this.text, required this.isDark});
  final String text;
  final bool isDark;

  @override
  Widget build(BuildContext context) {
    return Container(
      margin: const EdgeInsets.only(bottom: 8),
      padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 2),
      decoration: BoxDecoration(
        color: isDark ? Colors.white.withValues(alpha: 0.1) : NemoColors.gray100,
        borderRadius: BorderRadius.circular(10),
      ),
      child: Text(
        text,
        style: TextStyle(
          fontSize: 10,
          fontWeight: FontWeight.bold,
          color: isDark ? NemoColors.darkTextSecondary : NemoColors.gray500,
          letterSpacing: 1,
        ),
      ),
    );
  }
}

class _Label extends StatelessWidget {
  const _Label(this.text, {required this.color});
  final String text;
  final Color color;

  @override
  Widget build(BuildContext context) {
    return Text(
      text,
      style: TextStyle(
        fontSize: 12,
        fontWeight: FontWeight.w500, // Medium
        color: color,
        letterSpacing: 1,
      ),
    );
  }
}

class _ExampleRow extends StatelessWidget {
  const _ExampleRow({
    required this.japanese,
    this.chinese,
    required this.id,
    this.onSpeak,
    this.isPlaying = false,
    required this.primaryTextColor,
    required this.secondaryTextColor,
  });

  final String japanese;
  final String? chinese;
  final String id;
  final Function(String, String, String)? onSpeak;
  final bool isPlaying;
  final Color primaryTextColor;
  final Color secondaryTextColor;

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.only(bottom: 16),
      child: Row(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Expanded(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                NemoFuriganaText(
                  text: japanese,
                  baseTextStyle: TextStyle(
                    fontSize: 15,
                    height: 1.6,
                    color: primaryTextColor,
                    fontWeight: FontWeight.normal,
                  ),
                ),
                if (chinese != null) ...[
                  const SizedBox(height: 2),
                  Text(
                    chinese!,
                    style: TextStyle(fontSize: 12, color: secondaryTextColor),
                  ),
                ],
              ],
            ),
          ),
          if (onSpeak != null)
            IconButton(
              onPressed: () => onSpeak!(japanese, chinese ?? '', id),
              icon: Icon(
                isPlaying ? Icons.volume_up_rounded : Icons.volume_down_rounded,
                color: secondaryTextColor.withValues(alpha: 0.6),
                size: 20,
              ),
              padding: EdgeInsets.zero,
              constraints: const BoxConstraints(),
            ),
        ],
      ),
    );
  }
}

class _SpeakerButton extends StatelessWidget {
  const _SpeakerButton({
    required this.onPressed,
    required this.isPlaying,
    required this.color,
    required this.backgroundColor,
  });

  final VoidCallback onPressed;
  final bool isPlaying;
  final Color color;
  final Color backgroundColor;

  @override
  Widget build(BuildContext context) {
    return Container(
      width: 48,
      height: 48,
      decoration: BoxDecoration(color: backgroundColor, shape: BoxShape.circle),
      child: IconButton(
        onPressed: onPressed,
        icon: Icon(isPlaying ? Icons.volume_up_rounded : Icons.volume_down_rounded, color: color, size: 26),
        padding: EdgeInsets.zero,
      ),
    );
  }
}

class _PracticeButton extends StatelessWidget {
  const _PracticeButton({
    required this.onPressed,
    required this.color,
    required this.backgroundColor,
  });

  final VoidCallback onPressed;
  final Color color;
  final Color backgroundColor;

  @override
  Widget build(BuildContext context) {
    return InkWell(
      onTap: onPressed,
      borderRadius: BorderRadius.circular(24),
      child: Container(
        width: 48,
        height: 48,
        decoration: BoxDecoration(color: backgroundColor, shape: BoxShape.circle),
        alignment: Alignment.center,
        child: Icon(Icons.translate_rounded, color: color, size: 26),
      ),
    );
  }
}

File: archive/feature/learning/src/main/java/com/jian/nemo/feature/learning/presentation/components/cards/SRSLearningCard.kt
package com.jian.nemo.feature.learning.presentation.components.cards

import androidx.compose.animation.AnimatedVisibility
import androidx.compose.animation.core.tween
import androidx.compose.animation.fadeIn
import androidx.compose.animation.slideInVertically
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.rounded.Translate
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.blur
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.luminance
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.PlatformTextStyle
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.jian.nemo.core.domain.model.Word
import com.jian.nemo.core.designsystem.theme.NemoSurfaceCard
import com.jian.nemo.core.designsystem.theme.NemoSurfaceCardDark
import androidx.compose.ui.graphics.painter.ColorPainter
import com.jian.nemo.core.designsystem.theme.NemoNeutrals
import com.jian.nemo.core.ui.component.text.FuriganaText
import coil.compose.AsyncImage
import coil.decode.SvgDecoder
import coil.request.ImageRequest
import androidx.compose.ui.platform.LocalContext
import com.jian.nemo.core.designsystem.theme.IosColors
import com.jian.nemo.core.ui.component.speaker.SpeakerButton
import com.jian.nemo.feature.learning.presentation.CardBadge
import kotlin.math.abs

/**
 * HIG 风格配色列表 - 用于生成随机但稳定的颜色
 */
private val higColors = listOf(
    IosColors.Blue,
    IosColors.Green,
    IosColors.Orange,
    IosColors.Pink,
    IosColors.Indigo,
    IosColors.Purple,
    IosColors.Mint,
    IosColors.Red,
    IosColors.Teal,
    IosColors.Yellow
)

/**
 * 获取随机贴纸文件名
 */
private fun getStickerForWord(wordId: Int): String {
    val stickers = listOf(
        "bad_taste", "birthday", "cleaning", "confused", "cooking",
        "cool", "eating_noodles", "headache", "listening_music", "love",
        "pretend_sleep", "really", "receiving_gift", "scared", "selfie",
        "shocked", "singing", "something", "studying", "superman",
        "sure", "taking_photo", "waving", "yoga", "zoning_out"
    )
    val index = abs(wordId) % stickers.size
    return stickers[index]
}

/**
 * 根据单词 ID 获取稳定的随机颜色
 */
private fun getColorForWord(wordId: Int): Color {
    val index = abs(wordId) % higColors.size
    return higColors[index]
}

/**
 * @param word 单词数据
 * @param isAnswerShown 是否显示答案
 * @param onPracticeClick 点击跟打练习按钮的回调
 * @param onSpeakWord 朗读单词回调
 * @param onSpeakExample 朗读例句回调 (日文, 中文, ID)
 * @param playingAudioId 当前正在朗读的 ID
 */
@Composable
fun SRSLearningCard(
    word: Word,
    isAnswerShown: Boolean,
    cardBadge: CardBadge? = null,
    modifier: Modifier = Modifier,
    onPracticeClick: (() -> Unit)? = null,
    onSpeakWord: (() -> Unit)? = null,
    onSpeakExample: ((String, String, String) -> Unit)? = null,
    playingAudioId: String? = null
) {
    // 检测深色模式
    val isDarkTheme = MaterialTheme.colorScheme.background.luminance() < 0.5

    // 根据主题选择颜色
    val cardBackground = if (isDarkTheme) NemoSurfaceCardDark else NemoSurfaceCard

    // Border/Divider
    // Border/Divider - Refined to 0.5dp for premium feel
    val borderColor = if (isDarkTheme) MaterialTheme.colorScheme.outlineVariant.copy(alpha = 0.1f) else MaterialTheme.colorScheme.outlineVariant.copy(alpha = 0.2f)
    val dividerColor = if (isDarkTheme) Color.White.copy(alpha = 0.1f) else NemoNeutrals.Gray100

    // Shadow Parameters
    val shadowElevation = if (isDarkTheme) 2.dp else 10.dp
    val shadowColor = if (isDarkTheme) Color.Black.copy(alpha = 0.4f) else Color.Black.copy(alpha = 0.03f)

    // Text: 使用 NemoNeutrals 定义的精准色值
    val primaryTextColor = if (isDarkTheme) NemoNeutrals.DarkTextPrimary else NemoNeutrals.Gray900
    val secondaryTextColor = if (isDarkTheme) NemoNeutrals.Gray400 else NemoNeutrals.Gray400

    // Labels
    val labelBgColor = if (isDarkTheme) Color.White.copy(alpha = 0.1f) else NemoNeutrals.Gray100
    val labelTextColor = if (isDarkTheme) NemoNeutrals.DarkTextSecondary else NemoNeutrals.Gray500
    val hiraganaColorHidden = if (isDarkTheme) Color.White.copy(alpha = 0.2f) else NemoNeutrals.Gray300

    // 跟打按钮颜色 - 根据单词 ID 生成稳定的随机颜色
    val practiceButtonColor = remember(word.id) { getColorForWord(word.id) }
    val practiceButtonBgColor = if (isDarkTheme) {
        practiceButtonColor.copy(alpha = 0.2f)
    } else {
        practiceButtonColor.copy(alpha = 0.1f)
    }

    Column(
        modifier = modifier
            .fillMaxWidth()
            .verticalScroll(rememberScrollState())
            .padding(bottom = 100.dp)
    ) {
        // --- Question Area ---
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .shadow(
                    elevation = shadowElevation,
                    shape = RoundedCornerShape(26.dp),
                    spotColor = shadowColor,
                    ambientColor = shadowColor
                )
                .background(cardBackground, RoundedCornerShape(26.dp))
                .border(0.5.dp, borderColor, RoundedCornerShape(26.dp))
                .padding(vertical = 24.dp, horizontal = 16.dp)
        ) {
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(top = if (cardBadge != null) 20.dp else 0.dp),
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                // Japanese
                Text(
                    text = word.japanese,
                    color = primaryTextColor,
                    fontSize = 48.sp,
                    fontWeight = FontWeight.ExtraBold, // Refined from Black to ExtraBold
                    letterSpacing = (-1).sp,
                    modifier = Modifier.padding(bottom = 8.dp),
                    textAlign = TextAlign.Center,
                    lineHeight = 56.sp,
                    style = TextStyle(
                        platformStyle = PlatformTextStyle(
                            includeFontPadding = false
                        )
                    )
                )

                // Hiragana
                Text(
                    text = word.hiragana,
                    color = if (isAnswerShown) NemoNeutrals.Blue600 else hiraganaColorHidden,
                    fontSize = 20.sp,
                    fontWeight = FontWeight.Medium,
                    modifier = Modifier.then(
                        if (!isAnswerShown) Modifier.blur(8.dp) else Modifier
                    )
                )
            }

            cardBadge?.let { badge ->
                val (text, bgColor, textColor) = when (badge) {
                    CardBadge.NEW -> Triple("新学", if (isDarkTheme) Color(0xFF1E3A8A) else Color(0xFFE0EDFF), if (isDarkTheme) Color(0xFFBFDBFE) else Color(0xFF1D4ED8))
                    CardBadge.REVIEW -> Triple("复习", if (isDarkTheme) Color(0xFF14532D) else Color(0xFFDCFCE7), if (isDarkTheme) Color(0xFFBBF7D0) else Color(0xFF166534))
                    CardBadge.RELEARN -> Triple("重学", if (isDarkTheme) Color(0xFF7C2D12) else Color(0xFFFFEDD5), if (isDarkTheme) Color(0xFFFED7AA) else Color(0xFF9A3412))
                }

                Box(
                    modifier = Modifier
                        .align(Alignment.TopEnd)
                        .background(bgColor, CircleShape)
                        .padding(horizontal = 14.dp, vertical = 6.dp)
                ) {
                    Text(
                        text = text,
                        color = textColor,
                        fontSize = 12.sp,
                        fontWeight = FontWeight.Black,
                        letterSpacing = 1.2.sp
                    )
                }
            }
        }

        Spacer(modifier = Modifier.height(16.dp))

        // --- Sticker Area (答案显示前) ---
        if (!isAnswerShown) {
            val context = LocalContext.current
            val stickerName = remember(word.id) { getStickerForWord(word.id) }

            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(vertical = 24.dp),
                contentAlignment = Alignment.Center
            ) {
                AsyncImage(
                    model = ImageRequest.Builder(context)
                        .data("file:///android_asset/stickers/${stickerName}.svg")
                        .decoderFactory(SvgDecoder.Factory())
                        .crossfade(300)
                        .build(),
                    contentDescription = null,
                    modifier = Modifier.size(320.dp),
                    placeholder = ColorPainter(Color.Gray.copy(alpha = 0.1f)),
                    error = ColorPainter(Color.Red.copy(alpha = 0.05f))
                )
            }
        }

        // --- Answer Area ---
        AnimatedVisibility(
            visible = isAnswerShown,
            enter = fadeIn(tween(300)) + slideInVertically(tween(300)) { it / 2 }
        ) {
            Box(
                modifier = Modifier.fillMaxWidth()
            ) {
                Column(
                    modifier = Modifier
                        .fillMaxWidth()
                        .shadow(
                            elevation = shadowElevation,
                            shape = RoundedCornerShape(26.dp),
                            spotColor = shadowColor,
                            ambientColor = shadowColor
                        )
                        .background(cardBackground.copy(alpha = 0.95f), RoundedCornerShape(26.dp)) // Slightly increased opacity
                        .border(0.5.dp, borderColor, RoundedCornerShape(26.dp))
                        .padding(24.dp)
                ) {
                    // Meaning
                    Column(modifier = Modifier.padding(bottom = 16.dp)) {
                        // POS Tag
                        Box(
                            modifier = Modifier
                                .padding(bottom = 8.dp)
                                .background(labelBgColor, CircleShape)
                                .padding(horizontal = 8.dp, vertical = 2.dp)
                        ) {
                            Text(
                                text = word.pos ?: "未知",
                                color = labelTextColor,
                                fontSize = 10.sp,
                                fontWeight = FontWeight.Bold,
                                letterSpacing = 1.sp
                            )
                        }

                        Text(
                            text = "含义",
                            color = secondaryTextColor,
                            fontSize = 12.sp,
                            fontWeight = FontWeight.Medium, // Refined from Bold to Medium
                            letterSpacing = 1.sp,
                            modifier = Modifier.padding(bottom = 4.dp)
                        )
                        Text(
                            text = word.chinese,
                            color = primaryTextColor,
                            fontSize = 24.sp,
                            fontWeight = FontWeight.Bold
                        )
                    }

                    // Divider
                    Box(
                        modifier = Modifier
                            .fillMaxWidth()
                            .height(1.dp)
                            .background(dividerColor)
                    )

                    Spacer(modifier = Modifier.height(16.dp))

                    // Example
                    Column {
                        Text(
                            text = "例句",
                            color = secondaryTextColor,
                            fontSize = 12.sp,
                            fontWeight = FontWeight.Medium, // Refined from Bold to Medium
                            letterSpacing = 1.sp,
                            modifier = Modifier.padding(bottom = 8.dp)
                        )

                        val examples = listOf(
                            Triple(word.example1, word.gloss1, "example_${word.id}_1"),
                            Triple(word.example2, word.gloss2, "example_${word.id}_2"),
                            Triple(word.example3, word.gloss3, "example_${word.id}_3")
                        ).filter { !it.first.isNullOrBlank() }

                        if (examples.isNotEmpty()) {
                            examples.forEachIndexed { index, (example, gloss, id) ->
                                Row(
                                    modifier = Modifier.fillMaxWidth(),
                                    horizontalArrangement = Arrangement.SpaceBetween,
                                    verticalAlignment = Alignment.Top
                                ) {
                                    Column(modifier = Modifier.weight(1f)) {
                                        FuriganaText(
                                            text = example!!,
                                            baseTextStyle = MaterialTheme.typography.bodyLarge.copy(
                                                fontSize = 15.sp,
                                                fontWeight = FontWeight.Normal,
                                                lineHeight = 24.sp
                                            ),
                                            baseTextColor = primaryTextColor,
                                            furiganaTextSize = 9.sp,
                                            furiganaTextColor = secondaryTextColor,
                                            modifier = Modifier.padding(bottom = 4.dp)
                                        )
                                        if (!gloss.isNullOrBlank()) {
                                            Text(
                                                text = gloss,
                                                color = secondaryTextColor,
                                                fontSize = 12.sp,
                                                modifier = Modifier.padding(bottom = if (index < examples.lastIndex) 12.dp else 0.dp)
                                            )
                                        }
                                    }

                                    // 例句朗读按钮
                                    if (onSpeakExample != null && !example.isNullOrBlank()) {
                                        SpeakerButton(
                                            isPlaying = playingAudioId == id,
                                            onClick = { onSpeakExample(example, gloss ?: "", id) },
                                            tint = secondaryTextColor,
                                            size = 36.dp
                                        )
                                    }
                                }
                            }
                        } else {
                            Text(
                                text = "暂无例句",
                                color = secondaryTextColor,
                                fontSize = 14.sp
                            )
                        }
                    }
                }

                // 按钮区域（右上角）：朗读 + 跟打练习
                Row(
                    modifier = Modifier
                        .align(Alignment.TopEnd)
                        .padding(12.dp),
                    horizontalArrangement = Arrangement.spacedBy(8.dp)
                ) {
                    // 朗读单词按钮
                    if (onSpeakWord != null) {
                        SpeakerButton(
                            isPlaying = playingAudioId == "word",
                            onClick = { onSpeakWord() },
                            tint = practiceButtonColor,
                            size = 48.dp,
                            backgroundColor = practiceButtonBgColor
                        )
                    }

                    // 跟打练习按钮
                    if (onPracticeClick != null) {
                        Box(
                            modifier = Modifier
                                .size(48.dp)
                                .clip(CircleShape)
                                .background(practiceButtonBgColor)
                                .clickable { onPracticeClick() },
                            contentAlignment = Alignment.Center
                        ) {
                            Icon(
                                imageVector = Icons.Rounded.Translate,
                                contentDescription = "跟打练习",
                                tint = practiceButtonColor,
                                modifier = Modifier.size(26.dp)
                            )
                        }
                    }
                }
            }
        }
    }
}
